packages/aws-cdk-lib/custom-resources/lib/provider-framework/waiter-state-machine.ts

```diff
@@ -75,18 +153,34 @@ export class WaiterStateMachine extends Construct {
       },
     });
 
+    const logOptions = logGroup ? {
+      LoggingConfiguration: {
+        Destinations: [{
+          CloudWatchLogsLogGroup: {
+            LogGroupArn: logGroup.logGroupArn,
+          },
+        }],
+        IncludeExecutionData: props.logOptions?.includeExecutionData ?? false,
+        Level: props.logOptions?.level ?? LogLevel.ERROR,
+      },
+    } : undefined;
+
     const resource = new CfnResource(this, 'Resource', {
       type: 'AWS::StepFunctions::StateMachine',
       properties: {
         DefinitionString: definition,
         RoleArn: role.roleArn,
+        ...logOptions,
```

Comment:
Thanks, I will change like this.

However I will use `if (disableLogging) return undefined;` instead of `if (!logOptions || disableLogging) return undefined;`. Because it needs to create logs and policies by default even if `logOptions` is not specified, according to this issue.

Reply:
And I think a return value type of the method cannot be `CfnStateMachine.LoggingConfigurationProperty`.

Because `AWS::StepFunctions::StateMachine` is created by not L1 Construct for `StateMachine` but `CfnResource`.

So I will return by not specifying type for now.

https://github.com/aws/aws-cdk/pull/27310/files#diff-ba99cb0b24f721e7bd6405b5f84ba44cd64f95932fa1f0974d7676855680a25bR168-R172

Reply:
Or how about using `CfnStateMachine` instead of `CfnResource` as follows?

```ts
    const resource = new CfnStateMachine(this, 'Resource', {
      definitionString: definition,
      roleArn: role.roleArn,
      loggingConfiguration: this.renderLoggingConfiguration(role, props.logOptions, props.disableLogging),
    });
```

Reply:
I think that the current implementation should be fine.
Thanks for the clarification and the adjustment on the `if` condition.